### ISSUE TITLE
.install callback swallowing console log?

### ISSUE THREAD

**Issue by watsontom100** [getsentry/raven-node#255]
I am trying to log to the console the error that is being caught by the global handler:

```
Raven.install(function(called, err) {
    console.error(err);
    process.exit(1);
});

Raven.context(function() {
    throw new Error("Hello World");
});
```

The process.exit(1) runs however nothing prints to the console (nothing prints when I change this to `console.log("test")` either). I presumed it would run similar to patchGlobal:

```
client.patchGlobal(function(called, err) {
    console.error(err);
    process.exit(1);
});
```

The docs seem to suggest you can log things at this point?

**Comment by LewisJEllis** [comment]
I'm assuming you did `Raven.config(dsn)` prior to the install etc - if not that would also cause problems.

There's actually a bit of a subtlety going on here. Right now the `context` function captures your error but then (incorrectly) swallows the error as described in #257, so the global handler from `install` never actually runs; the script just runs to completion and exits. The `process.exit(1)` isn't being called - check your exit code, it'll be 0 not 1. That's why you aren't seeing the `console.error` print out, but you would with:
```javascript
Raven.install(function(called, err) {
    console.error(err);
    process.exit(1);
});

throw new Error("Hello World");
```
because then the error doesn't get captured+swallowed by the context wrapper. I don't think there's anything wrong with the `install` callback, it's just that we're never making it to that callback. That'll be fixed by whatever resolution is reached in #257, but for now if you'd like to see when errors happen in the context, you can pass an `onErr` handler to `context`, maybe something like:
```javascript
Raven.install(function(called, err) {
    console.error(err);
    process.exit(1);
});

Raven.context(function () {
    throw new Error("Hello World");
}, function (err) {
    Raven.captureException(err, function () {
        console.error(err);
        process.exit(1);
    });
});
```
This is a bit manual, but gives flexibility to do whatever you might need. You can see the default `onErr` [here](https://github.com/getsentry/raven-node/blob/master/lib/client.js#L282).

There's also the `logged` event you could use to know when we've captured an error:
```javascript
Raven.on('logged', function (eventId) {
    ...
});
```
but note that this is any capture; not all captures necessarily hit the global handler or warrant a process shutdown.

I'm gonna close this as falling under the scope of #257, but feel free to continue discussion, or reopen if something doesn't seem to line up.

**Comment by watsontom100** [comment]
Yep, definitely falls under that scope, cheers! Might be worth updating the docs in the meantime as it currently shows an example of the install(cb) pattern and how it can be used.
